tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1926,3 +1927,158 @@ def call(self, inputs, state):
+
+    if num_proj:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_proj))
+      self._output_size = num_proj
+    else:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_units))
+      self._output_size = num_units
+
+  @property
+  def state_size(self):
+    return self._state_size
+
+  @property
+  def output_size(self):
+    return self._output_size
+
+  def _get_input_for_group(self, inpt, group_id, group_size):
+    """Slices inputs into groups to prepare for processing by cell's groups
+
+    Args:
+      inpt: inputs
+      group_id: group id, for which to prepare extract input_group_id
+      group_size: size of the group
+
+    Returns:
+      subset of inputs corresponding to group "group_id"
+    """
+    return array_ops.slice(input_=inpt,
+                           begin=[0, group_id * group_size],
+                           size=[inpt.get_shape()[0].value, group_size],
+                           name="GLSTMinputGroupCreation")
+
+  def __call__(self, inputs, state):
+    """Run one step of G-LSTM.
+
+    Args:
+      inputs: input Tensor, 2D, batch x num_units.
+      state: this must be a tuple of state Tensors, both `2-D`, with column sizes `c_state` and `m_state`.
+
+    Returns:
+      A tuple containing:
+
+      - A `2-D, [batch x output_dim]`, Tensor representing the output of the
+        G-LSTM after reading `inputs` when previous state was `state`.
+        Here output_dim is:
+           num_proj if num_proj was set,
+           num_units otherwise.
+      - Tensor(s) representing the new state of G-LSTM after reading `inputs` when
+        the previous state was `state`.  Same type and shape(s) as `state`.
+
+    Raises:
+      ValueError: If input size cannot be inferred from inputs via
+        static shape inference.
+    """
+    (c_prev, m_prev) = state
+
+    input_size = inputs.get_shape().with_rank(2)[1]
+    if input_size.value is None:
+      raise ValueError("Could not infer input size from inputs.get_shape()[-1]")
+    dtype = inputs.dtype
+    with vs.variable_scope("glstm_cell",
+                           initializer=self._initializer):
+      i_parts = []
+      j_parts = []
+      f_parts = []
+      o_parts = []
+
+      for group_id in xrange(self._number_of_groups):
+        with vs.variable_scope("group%d"%group_id):
+          x_g_id = array_ops.concat([self._get_input_for_group(inputs, group_id, self._group_shape[0]),
+                                     self._get_input_for_group(m_prev, group_id, self._group_shape[0])], axis=1)
+          R_k = _linear(x_g_id, 4 * self._group_shape[1], bias=False)
+          i_k, j_k, f_k, o_k = array_ops.split(R_k, 4, 1)
+
+        i_parts.append(i_k)
+        j_parts.append(j_k)
+        f_parts.append(f_k)
+        o_parts.append(o_k)
+
+      #it is more efficient to have per total gate biases compared to per gate, per group biases
+      bi = vs.get_variable(name="biases_i",
```

Comment:
use "bias_i" instead of "biases_i" etc; we're going to be doing a global rename of all RNNCells to match keras naming conventions; they use "kernel" for the matrix and "bias" for the biases.